autogpt_platform/frontend/src/components/molecules/Dialog/Dialog.stories.tsx

```diff
@@ -0,0 +1,197 @@
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <p>This is a basic dialog with some content.</p>
+        <p>
+          It automatically adapts to screen size - modal on desktop, drawer on
+          mobile.
+        </p>
+      </Dialog.Content>
+    </Dialog>
+  );
+}
+
+function renderDialogWithoutTitle() {
+  return (
+    <Dialog>
+      <Dialog.Trigger>
+        <Button variant="primary">Open Dialog (no title)</Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <p className="flex min-h-[100px] flex-row items-center justify-center">
+          This dialog doesn&apos;t use the title prop, allowing for no header or
+          a custom header.
+        </p>
+      </Dialog.Content>
+    </Dialog>
+  );
+}
+
+function renderForceOpenDialog(args: any) {
+  return (
+    <Dialog {...args}>
+      <Dialog.Content>
+        <p>This dialog is forced open for preview purposes.</p>
+        <p>
+          In real usage, you&apos;d typically use a trigger or controlled state.
+        </p>
+      </Dialog.Content>
+    </Dialog>
+  );
+}
+
+function renderDialogWithFooter() {
+  return (
+    <Dialog title="Dialog with Footer">
+      <Dialog.Trigger>
+        <Button variant="primary">Open Dialog with Footer</Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <p>This dialog includes a footer with action buttons.</p>
+        <p>Use the footer for primary and secondary actions.</p>
+        <Dialog.Footer>
+          <Button variant="ghost" size="small">
+            Cancel
+          </Button>
+          <Button variant="primary" size="small">
+            Confirm
+          </Button>
+        </Dialog.Footer>
+      </Dialog.Content>
+    </Dialog>
+  );
+}
+
+function renderControlledDialog() {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleToggle = async () => {
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <div className="space-y-4">
+      <Button variant="primary" onClick={handleToggle}>
+        {isOpen ? "Close" : "Open"} Controlled Dialog
+      </Button>
+
+      <Dialog
+        title="Controlled Dialog"
+        controlled={{
+          isOpen,
+          set: setIsOpen,
+        }}
+        onClose={() => console.log("Dialog closed")}
+      >
+        <Dialog.Content>
+          <div className="flex flex-col gap-4">
+            <p>This dialog is controlled by external state.</p>
+            <p>
+              Open state:{" "}
+              <span className="font-bold">{isOpen ? "Open" : "Closed"}</span>
+            </p>
+          </div>
+          <Button onClick={handleToggle} className="mt-8" size="small">
+            Close this modal
+          </Button>
+        </Dialog.Content>
+      </Dialog>
+    </div>
+  );
+}
+
+function renderCustomStyledDialog() {
+  return (
+    <Dialog
+      title="Custom Styled Dialog"
+      styling={{
+        maxWidth: "800px",
+        backgroundColor: "rgb(248 250 252)",
+        border: "2px solid rgb(59 130 246)",
```

Comment:
Are we not going to reuse this color?

Reply:
Noop this is just to demonstrate you can pass custom styles to the modal if needed 👍🏽